gstreamer-editing-services/src/lib.rs

```diff
@@ -0,0 +1,80 @@
+// Copyright (C) 2017 Sebastian Dröge <sebastian@centricular.com>
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+extern crate libc;
+
+use std::sync::{Once, ONCE_INIT};
+
+extern crate glib_sys as glib_ffi;
+extern crate gobject_sys as gobject_ffi;
+extern crate gio_sys as gio_ffi;
+extern crate gstreamer as gst;
+extern crate gstreamer_sys as gst_ffi;
+extern crate gstreamer_base as gst_base;
+extern crate gstreamer_base_sys as gst_base_ffi;
+extern crate gstreamer_pbutils_sys as gst_pbutils_ffi;
+extern crate gstreamer_pbutils as gst_pbutils;
+extern crate gstreamer_editing_services_sys as ffi;
+
+use glib::translate::{from_glib};
+
+#[macro_use]
+extern crate glib;
+extern crate gio;
+
+static GES_INIT: Once = ONCE_INIT;
+
+pub fn init() -> Result<(), &'static str> {
+    unsafe {
+        if from_glib(ffi::ges_init()) {
+            Ok(())
+        } else {
+            Err("Could not initialize GES.")
+        }
+    }
+}
+
+pub unsafe fn deinit() {
+    ffi::ges_deinit();
+}
+
+macro_rules! assert_initialized_main_thread {
+    () => {
+        if unsafe { ::gst_ffi::gst_is_initialized() } != ::glib_ffi::GTRUE {
+            panic!("GStreamer has not been initialized. Call `gst::init` first.");
+        }
+        ::GES_INIT.call_once(|| {
+            unsafe { ::ffi::ges_init() };
+        });
```

Comment:
Do we want lazy initialization of ges if there is also an explicit init function?

Reply:
If it can be done lazily and never fails anyway, sure

Reply:
It can, if user want to set some option he needs to do it manually, sounds safe to me.